js/script.js

```diff
@@ -1,6 +1,85 @@
+
+    // call this when the document is ready
+    // this function protects itself against being called more than once
+    function ready() {
+        if (!readyFired) {
+            // this must be set to true before we start calling callbacks
+            readyFired = true;
+            for (var i = 0; i < readyList.length; i++) {
+                // if a callback here happens to add new ready handlers,
+                // the docReady() function will see that it already fired
+                // and will schedule the callback to run right after
+                // this event loop finishes so all handlers will still execute
+                // in order and no new ones will be added to the readyList
+                // while we are processing the list
+                readyList[i].fn.call(window, readyList[i].ctx);
+            }
+            // allow any closures held by these functions to free
+            readyList = [];
+        }
+    }
+
+    function readyStateChange() {
+        if ( document.readyState === "complete" ) {
+            ready();
+        }
+    }
+
+    // This is the one public interface
+    // docReady(fn, context);
+    // the context argument is optional - if present, it will be passed
+    // as an argument to the callback
+    baseObj[funcName] = function(callback, context) {
+        if (typeof callback !== "function") {
+            throw new TypeError("callback for docReady(fn) must be a function");
+        }
+        // if ready has already fired, then just schedule the callback
+        // to fire asynchronously, but right away
+        if (readyFired) {
+            setTimeout(function() {callback(context);}, 1);
+            return;
+        } else {
+            // add the function and context to the list
+            readyList.push({fn: callback, ctx: context});
+        }
+        // if document already ready to go, schedule the ready function to run
+        if (document.readyState === "complete") {
+            setTimeout(ready, 1);
+        } else if (!readyEventHandlersInstalled) {
+            // otherwise if we don't have event handlers installed, install them
+            if (document.addEventListener) {
+                // first choice is DOMContentLoaded event
+                document.addEventListener("DOMContentLoaded", ready, false);
+                // backup is window load event
+                window.addEventListener("load", ready, false);
+            } else {
+                // must be IE
+                document.attachEvent("onreadystatechange", readyStateChange);
+                window.attachEvent("onload", ready);
+            }
+            readyEventHandlersInstalled = true;
+        }
+    }
+})("onReady", window);
+
+onReady(() => {
+    let roleChanger = document.getElementById("changeRole")
+
+    roleChanger.addEventListener("click", () => {
+        document.getElementById('roleTitle')
+            .innerHTML = "SEO Campaign Manager"    
+    });
 });
```

Comment:
The issue here was that the DOM had not fully initialized before the code was being called. This caused an error in the console and the event listener was never set properly. It worked when manually adding the code to the console because the DOM had already initialized at that point.

Any code that needs to interact directly with the DOM (or that is going to be called immediately) should be wrapped in a function and you should make certain that any globals that are called have been initialized properly. The `onReady` function here accomplishes that for the window and DOM, but there could be other things not covered by that function in more complex scripts or scenarios.

@KirklandGee Feel free to keep a discussion going if you have any questions or if you need any more help. You can also open a PR and request a review from me in the future if you want to go about that way. GH is great for collaboration and such.